### ISSUE TITLE
Refined queries should get a source class for the original query

### ISSUE THREAD

**Issue by christopherswenson** [looker-open-source/malloy#160]
Currently, the following query has no name and no source classes:

```malloy
query: flights -> airport_dashboard {where: origin.code: 'SJC'}
```

Therefore, when it renders, it renders as a table.

Instead, it should inherit the source class (or even the name outright) from the original query (here, `airport_dashboard`), and therefore render as a dashboard.

**Comment by mtoy-googly-moogly** [comment]
Yeah this is a problem with refined turtles and name matching. If we had tags and rendered based on tags and not names, this would not be a problem.
